examples/graphql-service-error-handling/graphql_service_error_handling.bal

```diff
@@ -0,0 +1,48 @@
+import ballerina/graphql;
+
+service class Profile {
+    private final int id;
+    private final string name;
+    private final int age;
+
+    function init(int id, string name, int age) {
+        self.id = id;
+        self.name = name;
+        self.age = age;
+    }
+
+    // This resource method does not have `nil` as a possible return type, which means the
+    // corresponding GraphQL field type is wrapped by GraphQL `NON_NULL` type. Therefore, if this
+    // method returns an `error`, the `name` value will become `null` and since it is a `NON_NULL`
+    // field, the value will be propagated to the upper level. This means the corresponding
+    // `profile` field value will become `null`.
+    resource function get name() returns string|error {
+        if self.id == 2 {
+            // return a mock error
+            return error("Error occurred while retrieving name");
+        }
+        return self.name;
+    }
+
+    // This resource method has `nil` as a possible return type, which means the corresponding
+    // GraphQL field type is nullable. Therefore, if this field returns an error, the field value
+    // can be `null`. Hence, when an error returned from this method,
+    resource function get age() returns int|error? {
+        if self.id == 1 {
+            // return a mock error
```

Comment:
```suggestion
            // it returns a mock error.
```

Reply:
Shouldn't this be 
```suggestion
            // Returns a mock error
```

to make it similar to the comment above?

Reply:
This seems to be a continuation of the previous comment.

Reply:
Updated.